Source/control.cpp

```diff
@@ -578,81 +581,64 @@ static void DrawFlask(const CelOutputBuffer &out, const CelOutputBuffer &celBuf,
 	out.BlitFromSkipColorIndexZero(celBuf, MakeSdlRect(sourcePosition.x, sourcePosition.y, FlaskWidth, h), targetPosition);
 }
 
-void DrawLifeFlask(const CelOutputBuffer &out)
+/**
+ * @brief Draws the part of the life/mana flasks potruding above the bottom panel
+ * @param out The display region to draw to
+ * @param sourceBuffer A sprite representing the appropriate background/empty flask style
+ * @param offset X coordinate offset for where the flask should be drawn
+ * @param fillPer How full the flask is (a value from 0 to 80)
+*/
+void DrawFlaskUpper(const CelOutputBuffer &out, const CelOutputBuffer &sourceBuffer, int offset, int fillPer)
 {
-	auto &myPlayer = plr[myplr];
-
-	int emptyPortion = 80 - myPlayer.UpdateHitPointPercentage();
-
-	// clamping because this function only draws the top 12% of the HP display
-	emptyPortion = clamp(emptyPortion, 0, 11) + 2; // +2 to account for the frame being included in the sprite
+	// clamping because this function only draws the top 12% of the flask display
+	int emptyPortion = clamp(80 - fillPer, 0, 11) + 2; // +2 to account for the frame being included in the sprite
 
 	// Draw the empty part of the flask
-	DrawFlask(out, pLifeBuff, { 13, 3 }, { PANEL_LEFT + 109, PANEL_TOP - 13 }, emptyPortion);
-	if (emptyPortion < 13) {
+	DrawFlask(out, sourceBuffer, { 13, 3 }, { PANEL_LEFT + offset, PANEL_TOP - 13 }, emptyPortion);
+	if (emptyPortion < 13)
 		// Draw the filled part of the flask
-		DrawFlask(out, pBtmBuff, { 109, emptyPortion + 3 }, { PANEL_LEFT + 109, PANEL_TOP - 13 + emptyPortion }, 13 - emptyPortion);
-	}
+		DrawFlask(out, pBtmBuff, { offset, emptyPortion + 3 }, { PANEL_LEFT + offset, PANEL_TOP - 13 + emptyPortion }, 13 - emptyPortion);
 }
 
-void UpdateLifeFlask(const CelOutputBuffer &out)
+void DrawFlaskLower(const CelOutputBuffer &out, const CelOutputBuffer &sourceBuffer, int offset, int fillPer)
 {
-	auto &myPlayer = plr[myplr];
-
-	int filled = myPlayer.UpdateHitPointPercentage();
+	int filled = clamp(fillPer, 0, 69);
 
-	if (filled > 69)
-		filled = 69;
-	else if (filled < 0)
-		filled = 0;
-	if (filled != 69)
-		DrawFlaskTop(out, { 96 + PANEL_X, PANEL_Y }, pLifeBuff, 16, 85 - filled);
-	if (filled != 0)
-		DrawPanelBox(out, { 96, 85 - filled, 88, filled }, { 96 + PANEL_X, PANEL_Y + 69 - filled });
+	if (filled < 69)
+		DrawFlaskTop(out, { PANEL_X + offset, PANEL_Y }, sourceBuffer, 16, 85 - filled);
+	if (filled > 0)
+		DrawPanelBox(out, { offset, 85 - filled, 88, filled }, { PANEL_X + offset, PANEL_Y + 69 - filled });
 }
 
-void DrawManaFlask(const CelOutputBuffer &out)
+// the Draw*Flask* functions use precalculated value of HP/ManaPer from an earlier call to control_update_life_mana()

```

Comment:
```suggestion
/**
 * the Draw*Flask* functions use precalculated value of HP/ManaPer from an earlier call to control_update_life_mana()
 */
```

Reply:
I wasn't sure if doxygen comments were appropriate here and it turns out VS picks this up instead of the header comment for the next function. I've updated the comments in the header instead to address this.

Reply:
Oh I thought this was a header comment for the next function, my bad :)